app/controllers/admin/bait_libraries/bait_library_suppliers_controller.rb

```diff
@@ -1,5 +1,5 @@
 class Admin::BaitLibraries::BaitLibrarySuppliersController < ApplicationController # rubocop:todo Style/Documentation
-  before_action :admin_login_required
+  authorize_resource class: BaitLibrary::Supplier
   before_action :discover_bait_library_supplier, only: %i[edit update destroy]
```

Comment:
Lots of this going on. Essentially switching out our role-our-own authorization for the can-can system. In this case we need to specify the class, as cancancan can't work it out from the controller.